.kokoro/common.sh

```diff
@@ -52,7 +54,105 @@ function retry_with_backoff {
     return $exit_code
 }
 
-## Helper functionss
+## Helper functions
 function now() { date +"%Y-%m-%d %H:%M:%S" | tr -d '\n'; }
 function msg() { println "$*" >&2; }
-function println() { printf '%s\n' "$(now) $*"; }
\ No newline at end of file
+function println() { printf '%s\n' "$(now) $*"; }
+
+function generate_modified_modules_list() {
+  # Find the files changed from when the PR branched to the last commit
+  # Filter for java modules and get all the unique elements
+  # grep returns 1 (error code) and exits the pipeline if there is no match
+  # If there is no match, it will return true so the rest of the commands can run
+  modified_files=$(git diff --name-only $KOKORO_GITHUB_PULL_REQUEST_COMMIT $KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH)
+  printf "Modified files:\n%s\n" "${modified_files}"
+
+  modified_module_list=()
+  # If root pom.xml is touched, run ITs on all the modules
+  root_pom_modified=$(echo "${modified_files}" | grep -e '^pom.xml$' || true)
+  if [[ ( -n $root_pom_modified ) || ( "${TEST_ALL_MODULES}" == "true" ) ]]; then
+    modules=$(mvn help:evaluate -Dexpression=project.modules | grep '<.*>.*</.*>' | sed -e 's/<.*>\(.*\)<\/.*>/\1/g')
+    for module in $modules; do
+      if [[ ! "${excluded_modules[*]}" =~ $module ]]; then
+        modified_module_list+=("${module}")
+      fi
+    done
+    echo "Testing the entire monorepo"
+  else
+    modules=$(echo "${modified_files}" | grep -e 'java-.*' || true)
+    printf "Files in java modules:\n%s\n" "${modules}"
+    if [[ -n $modules ]]; then
+      modules=$(echo "${modules}" | cut -d '/' -f1 | sort -u)
+      for module in $modules; do
+        modified_module_list+=("${module}")
+      done
+    else
+      echo "Found no changes in the java modules"
+    fi
+  fi
+}
+
+function assign_modules_to_job() {
+  modules_assigned_list=()
+  num=0
+  for module in "${modified_module_list[@]}"; do
+    # Add 1 as JOB_NUMBER is 1-indexed instead of 0-indexed
+    mod_num=$((num % NUM_JOBS + 1))
+    if [[ ! "${excluded_modules[*]}" =~ $module ]] && [[ $mod_num -eq $JOB_NUMBER ]]; then
+      modules_assigned_list+=("${module}")
+    fi
+    num=$((num + 1))
+  done
+}
+
+function run_graalvm_tests() {
+  printf "Running GraalVM ITs on:\n%s\n" "${module_list[*]}"
+  mvn -B ${INTEGRATION_TEST_ARGS} \
+    -pl "${module_list},!CoverageAggregator" \
+    -amd \
+    -ntp \
+    -DtrimStackTrace=false \
+    -Dclirr.skip=true \
+    -Denforcer.skip=true \
+    -Dcheckstyle.skip=true \
+    -Dflatten.skip=true \
+    -Danimal.sniffer.skip=true \
+    -Pnative \
+    -fae \
+    test
```

Comment:
Remove the -Penable-integration-tests since it only sets the skipITs user property to false (https://github.com/googleapis/java-shared-config/blob/abf28e47a74f94bc814e6ff21c7f832dce98c2eb/pom.xml#L682) which has no relation to the surefire plugin. May need to revisit this if graalvm supports integration tests with the failsafe plugin